elide-core/src/main/java/com/yahoo/elide/parsers/state/RelationshipTerminalState.java

```diff
@@ -84,7 +84,38 @@ public RelationshipTerminalState(PersistentResource record, String relationshipN
 
     @Override
     public Supplier<Pair<Integer, JsonNode>> handlePatch(StateContext state) {
-        return handleRequest(state, this::patch);
+        return handlePatchRequest(state, this::patch);
+    }
+
+    private Supplier<Pair<Integer, JsonNode>> handlePatchRequest(StateContext state,
+                                                            BiFunction<Data<Resource>, RequestScope, Boolean> handler) {
+        Data<Resource> data = state.getJsonApiDocument().getData();
+        handler.apply(data, state.getRequestScope());
+        return () -> Pair.of(
```

Comment:
This return statement is duplicated. Perhaps consider a small static method to encapsulate it. Since this logic appears to be identical, it's easier to maintain if we have it only in one place.